core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/screen/ScreenCollector.java

```diff
@@ -83,15 +97,62 @@ private boolean isPatternAndResultMD5Identical(byte[] screenshot) {
 
   @Override
   public void setParameters(Map<String, String> params) throws ParametersException {
-    //No parameters here
+    if (StringUtils.isNotBlank(params.get(XPATH_PARAM)) || StringUtils
+        .isNotBlank(params.get(CSS_PARAM))) {
+      setElementParams(params);
+      isPartial = true;
+    }
+  }
+
+  private byte[] takeScreenshot() throws ProcessingException {
+    try {
+      if (isPartial) {
+        SeleniumWaitHelper
+            .waitForElementToBePresent(webDriver, getLocator(), getTimeoutInSeconds());
+        WebElement element = webDriver.findElement(getLocator());
+        return getImagePart(getFullPageScreenshot(), element);
+      } else {
+        return getFullPageScreenshot();
+      }
+    } catch (NoSuchElementException e) {
+      throw new ProcessingException("Unable to find element for taking screenshot part", e);
+    } catch (IOException | WebDriverException e) {
+      throw new ProcessingException("Unable to take screenshot", e);
+    }
+
+  }
+
+  private byte[] getFullPageScreenshot() {
+    return ((TakesScreenshot) webDriver).getScreenshotAs(OutputType.BYTES);
   }
 
-  private byte[] takeScreenshot(WebDriver webDriver) throws ProcessingException {
+  private byte[] getImagePart(byte[] fullPage, WebElement webElement)
+      throws IOException, ProcessingException {
+    InputStream in = new ByteArrayInputStream(fullPage);
     try {
-      return ((TakesScreenshot) webDriver).getScreenshotAs(OutputType.BYTES);
-    } catch (WebDriverException e) {
-      throw new ProcessingException("Could not take screenshot, page is not loaded yet!", e);
+      BufferedImage fullImg = ImageIO.read(in);
+      Point point = webElement.getLocation();
+      Dimension size = webElement.getSize();
+      BufferedImage screenshotSection = fullImg.getSubimage(point.getX(), point.getY(),
+          size.getWidth(), size.getHeight());
+      return bufferedImageToByteArray(screenshotSection);
+    } catch (IOException e) {
+      throw new ProcessingException("Unable to create image from taken screenshot", e);
+    } finally {
+      IOUtils.closeQuietly(in);
     }
   }
 
+  private byte[] bufferedImageToByteArray(BufferedImage bufferedImage) throws ProcessingException {
+    ByteArrayOutputStream temporaryStream = new ByteArrayOutputStream();
+    try {
```

Comment:
You may get benefit from `Closable` interface that ByteArrayOutputStream implements:
```java
    try (ByteArrayOutputStream temporaryStream = new ByteArrayOutputStream()){
      ImageIO.write(bufferedImage, PNG_FORMAT, temporaryStream);
      temporaryStream.flush();
      return temporaryStream.toByteArray();
    } catch (IOException e) {
      throw new ProcessingException("Unable to convert screenshot part to byte Array", e);
    }
```